eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/DoctrineStorage.php

```diff
@@ -90,23 +90,32 @@ public function getNodePathString(VersionInfo $versionInfo)
      *
      * @param string $uri File IO uri (not legacy)
      * @param int $fieldId
+     * @param \eZ\Publish\SPI\Persistence\Content\VersionInfo $versionInfo
+     *
+     * @throws \eZ\Publish\Core\IO\Exception\InvalidBinaryFileIdException
```

Comment:
I'm not happy as well, but it might be thrown by `\eZ\Publish\Core\IO\UrlRedecoratorInterface::redecorateFromSource` (https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/IO/UrlRedecoratorInterface.php#L40). I can catch it, convert, and re-throw as a `\eZ\Publish\API\Repository\Exceptions\InvalidArgumentException` if you see it important. 

Simply speaking, I just added a `@thrown` tag which has been missing before I introduced the changes I did.

Reply:
> I'm not happy as well, but it might be thrown by `\eZ\Publish\Core\IO\UrlRedecoratorInterface::redecorateFromSource` (https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/IO/UrlRedecoratorInterface.php#L40). I can catch it, convert, and re-throw as a `\eZ\Publish\API\Repository\Exceptions\InvalidArgumentException` if you see it important.
> 
> Simply speaking, I just added a `@thrown` tag which has been missing before I introduced the changes I did.

@kmadejski my point was that `\eZ\Publish\Core\IO\Exception\InvalidBinaryFileIdException` is an instance of `\eZ\Publish\API\Repository\Exceptions\InvalidArgumentException`, so it would be perfectly fine to declare it like that on the abstract `Gateway` :-)

Reply:
> @kmadejski my point was that \eZ\Publish\Core\IO\Exception\InvalidBinaryFileIdException is an instance of \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException, so it would be perfectly fine to declare it like that on the abstract Gateway :-)

I totally agree. I don't know how I could miss that. 🤦‍♂ I think it's good to keep it as an API exception everywhere. Done in https://github.com/ezsystems/ezpublish-kernel/pull/2835/commits/17bc48ffdab9045c1429abf548e15e1ce2089dd2.